package.json

```diff
@@ -31,6 +31,7 @@
     "@angular/core": "^4.0.0",
     "@angular/forms": "^4.0.0",
     "@angular/http": "^4.0.0",
+    "@angular/router": "^4.0.0",
```

Comment:
In JIT mode that might work, but it won't be the case for AoT. Angular uses the type for DI, so the generated code actually *does* have a real reference to the symbol. 

Trying it out now, if I create an empty project w/ the CLI, add `MdIconModule` and delete `@angular/http` from node_modules, it doesn't even build:
```
ERROR in Error encountered resolving symbol values statically. 
Could not resolve @angular/http relative to 
/tmp/deptest/node_modules/@angular/material/typings/index.d.ts., 
resolving symbol ICON_REGISTRY_PROVIDER in 
/tmp/deptest/node_modules/@angular/material/typings/index.d.ts, 
resolving symbol MdIconModule in 
/tmp/deptest/node_modules/@angular/material/typings/index.d.ts, 
resolving symbol MdIconModule in 
/tmp/deptest/node_modules/@angular/material/typings/index.d.ts
```
(which is an error coming from the metadata extractor)